platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,826 @@
+consensus nodes. See the TSS-Library proposal for more details on the Groth21.
+
+The Groth21 algorithm uses [Shamir Secret Sharing](https://en.wikipedia.org/wiki/Shamir%27s_secret_sharing) to hide the
+private key of the ledger and distribute shares of it so that a threshold number of node signatures on a message can
+generate the ledger signature on the same message. The threshold should be high enough to ensure at least 1 honest
+share is required to participate in the signing since a threshold number of shares can also collude to recover the
+ledger private key, if they so choose.
+
+The Groth21 algorithm requires specific Elliptic Curve (EC)s with the ability to produce bilinear pairings. Each
+node will need its own long-term EC key pair on the curve for use in the groth21 TSS algorithm. Each node receives
+some number of shares in proportion to the consensus weight assigned to it. Each share is an EC key pair on the
+same curve. All public keys of the shares are known, and aggregating a threshold number of share public keys will
+produce the ledger id. Only the node has access to the private key of its own shares. Each node must keep their
+private keys to themselves since a threshold number of share private keys can aggregate to recover the ledger
+private key. An aggregate number of signatures on the same message from the share private keys will aggregate to
+recover the ledger signature on the message.
+
+Transferring the ability to generate ledger signatures from one set of consensus nodes to another is done by having
+each node generate a `TssMessage` for each share that fractures the share into a number of subshares or `shares of
+shares`. Each `share of a share` is encrypted with the public key of the node in the next consensus roster that it
+belongs to so that only the intended node can use the `share of a share` to recover that node's share's private key.
+The collection of `TssMessages` generated from nodes in the previous consensus roster forms the `key material` for
+the new consensus roster. To bootstrap a network and generate the ledger private and public keys, each node creates
+a random share for themselves and generates a `TssMessage` containing `shares of shares` for the total number of shares
+needed in the consensus roster. The use of random shares at the outset creates a random ledger private key that nobody
+knows.
+
+![TSS re-keying next consensus roster](./Groth21-re-keying-shares-of-shares.drawio.svg)
+
+This process of re-keying the next consensus roster during an address book change takes an asynchronous amount of
+time through multiple rounds of consensus to complete. This requires saving incremental progress in the state to ensure
+that the process can resume from any point if a node or the network restarts. Switching to the next consensus roster
+cannot happen until that roster has enough nodes able to recover a threshold number of shares so that an aggregation of
```

Comment:
> until that roster has enough nodes able to recover a threshold number of shares

I don't think I understand this statement. Since a TSS message contains data bound for each and every share owner, either *everyone* can recover their shares, or *no one* can.

Reply:
I've modified the statement for clarity.  Please re-review it when I push it.